src/gui/EditWidgetIcons.cpp

```diff
@@ -120,6 +130,136 @@ void EditWidgetIcons::load(Uuid currentUuid, Database* database, IconStruct icon
     }
 }
 
+void EditWidgetIcons::setUrl(const QString &url)
+{
+    m_url = QUrl(url);
+    // We do not want to send any user information to google
+    m_ui->faviconButton->setVisible(m_url.userInfo().isEmpty() && (m_url.scheme() == "http" || m_url.scheme() == "https"));
+    abortFaviconDownload();
+}
+
+void EditWidgetIcons::downloadFavicon()
+{
+    // We do not want to send any user information to google
+    Q_ASSERT(m_url.userInfo().isEmpty());
+    m_progress = new QProgressDialog(this);
```

Comment:
Could the setting QProgressDialog code (line 145 - 152) be move after line 176?
Then we don't need to faked the Maximum value to 10 and the value won't change from 10 to real request value suddenly.